cmd/legacy_param_mapper.go

```diff
@@ -150,6 +150,7 @@ func PopulateNewConfigFromLegacyFlagsAndConfig(c cliContext, flags *flagStorage,
 	if err := cfg.ValidateConfig(resolvedConfig); err != nil {
 		return nil, err
 	}
+	cfg.Rationalize(resolvedConfig)
```

Comment:
Shall we do this before validation?
Same in root.go

Reply:
`ValidateConfig` ensures that the user-specified config is valid and `Rationalize` augments the config. 
* If `ValidateConfig` fails then user specified a wrong config; something that we don't support and err out.
* If `ValidateConfig` passes then `Rationalize` is just business logic to augment the config.

If we `ValidateConfig` after `Rationalize` then we are combining validation for both user-specified configuration as well as the augmented config which seems confusing.